--- Makefile.old	Fri Mar 10 21:59:28 2000
+++ Makefile	Mon Jul  1 00:12:13 2002
@@ -9,11 +9,11 @@
 	install -d -m 755 $(DESTDIR)/sbin
 	install -d -m 755 $(DESTDIR)/boot
 	install -d -m 755 $(DESTDIR)/etc
-	install -d -m 755 $(DESTDIR)/usr/man/man5
-	install -d -m 755 $(DESTDIR)/usr/man/man8
+	install -d -m 755 $(DESTDIR)/usr/share/man/man5
+	install -d -m 755 $(DESTDIR)/usr/share/man/man8
 	install -s quik/quik $(DESTDIR)/sbin
 	install -m 444 first/first.b second/second.b second/second $(DESTDIR)/boot
 	if [ -f $(DESTDIR)/etc/quik.conf ]; then :; \
 	  else set -x; install -m 644 etc/quik.conf $(DESTDIR)/etc; fi
-	install -m 444 man/quik.conf.5 $(DESTDIR)/usr/man/man5
-	install -m 444 man/quik.8 man/bootstrap.8 $(DESTDIR)/usr/man/man8
+	install -m 444 man/quik.conf.5 $(DESTDIR)/usr/share/man/man5
+	install -m 444 man/quik.8 man/bootstrap.8 $(DESTDIR)/usr/share/man/man8
